src/python/WMCore/REST/Server.py

```diff
@@ -836,6 +837,8 @@ def _call(self, param):
         # Format the response.
         response.headers['X-REST-Status'] = 100
         response.headers['Content-Type'] = format
+        if cherrypy.request.headers.get('Accept-Encoding', '') == 'gzip':
```

Comment:
You mentioned that without these lines, it won't work properly. But do we understand what happens with this etag stream? I need to educate myself on this, but you might be able to do further research to see what the options are.

Reply:
Please read WMCore/REST/Format.py `def stream_compress` code which divide data into chunks, etc. Therefore, we need to return entire object (in order to gzip it) before we get its chunks as this code doing.

Reply:
Without going deep into this code, it looks like `stream_compress` is actually compressing (and streaming?) data to the client. It supports a few compression libraries, as defined here:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/REST/Format.py#L490

so, I am inclined to say that the right way to provide this feature would be to define gzip as a new compress library and implement its own method under REST/Format.py, e.g. "_stream_compress_gzip".

Reply:
Alan, they answer is no. The Format.py takes individual records and compress them one by one and yields back to the client. While what I presented in this PR is gzip of entire `{'result". {'data'...}}` structure and send it as one blob back to the client. This are two different modes and I doubt they can be combined. The former assumes that client will read records one by one and decode them. The latter approach provide **ONE** gzipped object to the client which client needs to unzip. In former case a dedicated client is required, in latter it does not.

Please go deep into the code and decide how server should behave and which clients you require to deal with this server.